subsys/bluetooth/audio/shell/bap_broadcast_assistant.c

```diff
@@ -430,6 +490,177 @@ static int cmd_bap_broadcast_assistant_add_src(const struct shell *sh,
+		return true;
+	}
+
+	if (bt_uuid_cmp(&adv_uuid.uuid, BT_UUID_BROADCAST_AUDIO) != 0) {
+		return true;
+	}
+
+	broadcast_id = sys_get_le24(data->data + BT_UUID_SIZE_16);
+
+	if (broadcast_id != auto_scan.broadcast_id) {
+		/* Not the one we want */
+
+		return false;
+	}
+
+	shell_print(ctx_shell,
+		    "Found BAP broadcast source with addressand ID 0x%06X\n",
+		    broadcast_id);
+
+	bt_addr_le_copy(&param.addr, info->addr);
+	param.adv_sid = info->sid;
+	param.pa_interval = info->interval;
+	param.broadcast_id = broadcast_id;
+	param.pa_sync = auto_scan.pa_sync;
+	param.num_subgroups = 1;
+	param.subgroups = &auto_scan.subgroup;
+
+	err = bt_bap_broadcast_assistant_add_src(default_conn, &param);
+	if (err) {
+		shell_print(ctx_shell, "Failed to add source: %d", err);
+	}
+
+	memset(&auto_scan, 0, sizeof(auto_scan));
+	auto_scan.broadcast_id = INVALID_BROADCAST_ID;
+
+	return false;
+}
+
+static void scan_recv_cb(const struct bt_le_scan_recv_info *info,
+			 struct net_buf_simple *ad)
+{
+	if (auto_scan.broadcast_id == INVALID_BROADCAST_ID) {
+		/* no op */
+		return;
+	}
+
+	/* We are only interested in non-connectable periodic advertisers */
+	if ((info->adv_props & BT_GAP_ADV_PROP_CONNECTABLE) != 0 ||
+	    info->interval == 0) {
+		return;
+	}
+
+	bt_data_parse(ad, broadcast_source_found, (void *)info);
+}
+
+static void scan_timeout_cb(void)
+{
+	shell_print(ctx_shell, "Scan timeout");
+
+	if (auto_scan.broadcast_id != INVALID_BROADCAST_ID) {
+		memset(&auto_scan, 0, sizeof(auto_scan));
+		auto_scan.broadcast_id = INVALID_BROADCAST_ID;
+	}
+}
+
+static struct bt_le_scan_cb scan_callbacks = {
+	.recv = scan_recv_cb,
+	.timeout = scan_timeout_cb,
+};
+
+static int cmd_bap_broadcast_assistant_add_broadcast_id(const struct shell *sh,
+							size_t argc,
+							char **argv)
+{
+	struct bt_bap_scan_delegator_subgroup subgroup = { 0 };
+	static bool scan_cbs_registered;
+	unsigned long broadcast_id;
+	unsigned long pa_sync;
+	int err = 0;
+
+	if (!scan_cbs_registered) {
+		bt_le_scan_cb_register(&scan_callbacks);
+		scan_cbs_registered = true;
+	}
+
+	if (auto_scan.broadcast_id != INVALID_BROADCAST_ID) {
+		shell_info(sh, "Already scanning, wait for sync or timeout");
+
+		return -ENOEXEC;
+	}
+
+	broadcast_id = shell_strtoul(argv[0], 0, &err);
+	if (err != 0) {
+		shell_error(sh, "failed to parse broadcast_id: %d", err);
+
+		return -EINVAL;
+	} else if (broadcast_id > 0xFFFFFF /* 24 bits */) {
+		shell_error(sh, "Broadcast ID maximum 24 bits (was %x)",
+			    broadcast_id);
+
+		return -EINVAL;
+	}
+
+	pa_sync = shell_strtoul(argv[1], 0, &err);
+	if (err != 0) {
+		shell_error(sh, "failed to parse pa_sync: %d", err);
+
+		return -EINVAL;
+	} else if (pa_sync != 0U && pa_sync != 1U) {
+		shell_error(sh, "pa_sync shall be boolean: %ul", pa_sync);
+
+		return -EINVAL;
+	}
+
+	/* TODO: Support multiple subgroups */
+	if (argc > 2) {
+		subgroup.bis_sync = strtoul(argv[2], NULL, 0);
+		if (subgroup.bis_sync > UINT32_MAX) {
```

Comment:
if UINT32_MAX is the limit, maybe do a temp value from strtoull to check out of 32bit size or use errno to check

Reply:
Modified to use `shell_strtoul` like the other places where we get the `bis_sync`